realsense/hand/win/hand_module_object.cc

```diff
@@ -0,0 +1,611 @@
+// Copyright (c) 2016 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "realsense/hand/win/hand_module_object.h"
+
+// This file is auto-generated by hand_module.idl
+#include "hand_module.h" // NOLINT
+
+#include <vector>
+
+#include "base/bind.h"
+#include "base/logging.h"
+#include "base/strings/sys_string_conversions.h"
+#include "base/time/time.h"
+#include "realsense/common/win/common_utils.h"
+
+namespace realsense {
+namespace hand {
+
+#define PXC_SUCCEEDED(status) (((pxcStatus)(status)) >= PXC_STATUS_NO_ERROR)
+#define PXC_FAILED(status) (((pxcStatus)(status)) < PXC_STATUS_NO_ERROR)
+
+#define MESSAGE_TO_METHOD(message, method) \
+  handler_.Register( \
+      message, base::Bind(&method, base::Unretained(this)));
+
+using namespace realsense::common;  // NOLINT
+using namespace realsense::jsapi::hand_module;  // NOLINT
+using namespace xwalk::common;  // NOLINT
+
+inline void PopulatePoint2D(
+    Point2D* js_point_2d, const PXCPointF32& pxc_point_2d) {
+  js_point_2d->x = pxc_point_2d.x;
+  js_point_2d->y = pxc_point_2d.y;
+}
+
+inline void PopulatePoint3D(
+    Point3D* js_point_3d, const PXCPoint3DF32& pxc_point_3d) {
+  js_point_3d->x = pxc_point_3d.x;
+  js_point_3d->y = pxc_point_3d.y;
+  js_point_3d->z = pxc_point_3d.z;
+}
+
+inline void PopulatePoint4D(
+    Point4D* js_point_4d, const PXCPoint4DF32& pxc_point_4d) {
+  js_point_4d->x = pxc_point_4d.x;
+  js_point_4d->y = pxc_point_4d.y;
+  js_point_4d->z = pxc_point_4d.z;
+  js_point_4d->w = pxc_point_4d.w;
+}
+
+inline void PopulateJointData(
+    JointData* js_joint_data, const PXCHandData::JointData& pxc_joint_data) {
+  js_joint_data->confidence = pxc_joint_data.confidence;
+  PopulatePoint3D(&js_joint_data->position_world,
+                  pxc_joint_data.positionWorld);
+  PopulatePoint3D(&js_joint_data->position_image,
+                  pxc_joint_data.positionImage);
+  PopulatePoint4D(&js_joint_data->local_rotation, pxc_joint_data.localRotation);
+  PopulatePoint4D(&js_joint_data->global_orientation,
+                  pxc_joint_data.globalOrientation);
+  PopulatePoint3D(&js_joint_data->speed, pxc_joint_data.speed);
+}
+
+#define POPULATE_FINGER_JOINTS(Type, type, FINGER, finger) \
```

Comment:
Thanks. I don't find a better name either. How about we keep the current name and replace them once we get better idea?